### ISSUE TITLE
Servers returned by REST API users endpoint is a dictionary (not list) of dictionaries

### ISSUE THREAD

**Issue by rcthomas** [jupyterhub/jupyterhub#4036]
### Bug description

[JupyterHub's REST API docs](https://jupyterhub.readthedocs.io/en/stable/reference/rest-api.html) include an example response from [`GET /users`](https://jupyterhub.readthedocs.io/en/3.0.0/reference/rest-api.html#/default/get_users), with the value of `"servers"` depicted as a list of dictionaries.  However, my Hub's API returns it as a single dictionary of dictionaries, with the keys being the server names.  The `User` model at the bottom of the API spec also indicates that it should be a list of dictionaries.

The dictionary appears to be what's in the user model, at [apihandlers/base in APIHandler's user_model()](https://github.com/jupyterhub/jupyterhub/blob/3.0.0/jupyterhub/apihandlers/base.py#L328).

#### Expected behaviour

Actually I thought the dictionary response was normal behavior, I'd never really looked that close at the documentation before!  Is it just a documentation issue?

#### Actual behaviour

Got dictionary, not a list.

### How to reproduce

Configure JupyterHub to support named servers, start a few, and then do a `GET /users` as admin and I think you'll see a dictionary not a list.

**Comment by minrk** [comment]
Yes, just a documentation issue. user.servers has always been a dict; the docs should reflect that.

**Comment by hsadia538** [comment]
Hi, @minrk I want to work on the issue.

**Comment by hsadia538** [comment]
Hello,
I looked into the documentation and was able to find the Swagger specification generating the API documentation but still was not able to figure out where the example area is building from. `source/_static/rest-api.yml` has the complete Swagger with schema but examples are not defined here. Can you give me some pointers or ideas as to where should I look?

Thank You

**Comment by shree256** [comment]
Have raised a PR(#4047) that closes this issue. Please review.

**Comment by monil28** [comment]
I would like to work on this issue.

**Comment by Temidayo32** [comment]
@hsadia538 Are you still working on it?

**Comment by mahamtariq58** [comment]
I would like to work on this issue !

**Comment by hsadia538** [comment]
Hi all
I am not working on this anymore

**Comment by manics** [comment]
@hsadia538 Make sure you check-out the comments on the previous PR https://github.com/jupyterhub/jupyterhub/pull/4047
You'll need to do some investigation, or reach out to someone with more knowledge of swagger, so that you can define a dict-of-dict with variable field names.

**Comment by mahamtariq58** [comment]
@ Temidayo32  are you working on this issue?

**Comment by mahamtariq58** [comment]
@manics I am trying to solve this issue ..Can you guide me that is it just the documentation issue or coding issue?

**Comment by charlesjman** [comment]
Is anyone still working on this issue?

**Comment by manics** [comment]
@mahamtariq58 It's a documentation issue. https://github.com/jupyterhub/jupyterhub/pull/4047 made a start, but it's not that simple, as the response is a dictionary of server dictionaries.

**Comment by kamzzy** [comment]
I would like to work on this issue if no one is working on it

**Comment by consideRatio** [comment]
I reported this as well not finding this issue first

---

Here is a response from `/hub/api/users/consideratio` on a 3.1.1 hub, which [according to docs](https://jupyterhub.readthedocs.io/en/3.1.1/reference/rest-api.html#/default/get_users__name_) should be a list rather than a dictionary as provided in the response.

### Actual response: `servers` as a dictionary

```json
{
  "created": "2022-12-05T14:53:57.414129Z",
  "auth_state": null,
  "name": "consideratio",
  "groups": [],
  "server": null,
  "last_activity": "2023-03-02T10:10:55.591465Z",
  "kind": "user",
  "roles": [
    "admin",
    "user"
  ],
  "pending": null,
  "admin": true,
  "servers": {}
}
```

### Documented response: `servers` as a list

![image](https://user-images.githubusercontent.com/3837114/222400111-7e7890a6-0914-49de-a422-ec255f96df68.png)
